src/aind_behavior_experiment_launcher/behavior_launcher/slims_picker.py

```diff
@@ -0,0 +1,401 @@
+import logging
+import os
+from datetime import datetime
+from typing import List, Optional
+
+from aind_slims_api import SlimsClient, exceptions
+from aind_slims_api.models import SlimsBehaviorSession, SlimsInstrument, SlimsMouseContent, SlimsWaterlogResult
+from pydantic import ValidationError
+from typing_extensions import override
+
+import aind_behavior_experiment_launcher.ui as ui
+from aind_behavior_experiment_launcher.behavior_launcher._launcher import BehaviorLauncher, ByAnimalFiles
+from aind_behavior_experiment_launcher.launcher._base import TRig, TSession, TTaskLogic
+
+_BehaviorPickerAlias = ui.PickerBase[BehaviorLauncher[TRig, TSession, TTaskLogic], TRig, TSession, TTaskLogic]
+
+logger = logging.getLogger(__name__)
+
+try:
+    SLIMS_USERNAME = os.environ["SLIMS_USERNAME"]
+    SLIMS_PASSWORD = os.environ["SLIMS_PASSWORD"]
+except KeyError:
+    pass
+
+
+class SlimsPicker(_BehaviorPickerAlias[TRig, TSession, TTaskLogic]):
+    """
+    Picker class that handles the selection of rigs, sessions, and task logic from slims
+    """
+
+    def __init__(
+        self,
+        launcher: Optional[BehaviorLauncher] = None,
+        *,
+        ui_helper: Optional[ui.DefaultUIHelper] = None,
+        slims_url: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Initializes the picker with an optional launcher, UI helper, username, and password.
+
+        Args:
+            launcher (Optional[BehaviorLauncher]): The launcher instance.
+            ui_helper (Optional[_UiHelperBase]): The UI helper instance
+            slims_url(Optional[str]): slims url. Defaults to dev version of slims if not provided
+            username (Optional[str]): slims username. Defaults to SLIMS_USERNAME environment variable if not provided
+            password (Optional[str]): slims password. Default sto SLIMS_PASSWORD environment variable if not provided
+            **kwargs: Additional keyword arguments.
+        """
+
+        super().__init__(launcher, ui_helper=ui_helper, **kwargs)
+
+        self.slims_client = self.connect_to_slims(slims_url, username, password)
+        self.test_client_connection()
+
+        # initialize properties
+        self._slims_mouse = None
+        self._slims_session = None
+        self._slims_rig = None
```

Comment:
This private variables should be type-hinted if you already know their types here.